### PR TITLE
Provider user invitation wizard - part 2 - Saving new users

### PR DESCRIPTION
**Pull request by stevehook**
## Context

This PR is the second part of the implementation of a new wizard based design for the provider interface invite new user flow. It follows on from PR https://github.com/DFE-Digital/apply-for-teacher-training/pull/2523

This PR covers persistence of new and existing users to the database following input via the wizard. It doesn't cover the review page and change feature. It's behind a providers_can_manage_users_and_permissions feature flag and not yet linked to the users index page.

## Changes proposed in this pull request

- [x] Added a new `ProviderInterface#SaveProviderUserService` to implement the logic to either create or update a provider user based on state of the form model (`ProviderUserInvitationWizard `)
- [x] Removed the `ProviderUserInvitationWizard#save!` method - controller interacts directly with services to do persistence.
- [x] Integrate the new service with the existing `SaveAndInviteProviderUser` and `InviteProviderUser` services to implement the `ProviderUsersInvitationsController#commit` action.
- [x] Extend new system spec to assert that user is created in the database and that they get an invitation email.

## Guidance to review

- To some extent `ProviderInterface#SaveProviderUserService` is repeating work done by the existing `SaveProviderUser` service. I looked at reusing that but it seemed too tightly coupled to other parts to work with the new wizard.
- To get `SaveAndInviteProviderUser` and `InviteProviderUser` to work with the new service I had to extend `InviteProviderUser` to accept the `provider_user` as an email address as well as a model. This seemed like the easiest way to keep the various bits reasonably decoupled.

## Link to Trello card

https://trello.com/c/BcoX8Mt9/2431-implement-the-provider-user-invitation-wizard

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: spec/system/provider_interface/provider_invites_user_with_new_wizard_spec.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@
     when_i_visit_the_invite_user_wizard
     then_i_see_a_404_page
 
-    and_i_can_manage_users_for_a_provider
+    and_i_can_manage_users_for_two_provider
```
Comment (duncanjbrown) [inline spec/system/provider_interface/provider_invites_user_with_new_wizard_spec.rb:22]:
```suggestion
    and_i_can_manage_users_for_two_providers
```

**Comment by duncanjbrown** [comment]
This looks very good — just need to resolve the above design question

**Comment by duncanjbrown** [comment]
Just spoke to @adamsilver about this. His view is that the invite process should be additive — ie if we don't select things in this process they shouldn't be knocked out of an existing record. This seems sensible to me

**Comment by stevehook** [comment]
> Just spoke to @adamsilver about this. His view is that the invite process should be additive — ie if we don't select things in this process they shouldn't be knocked out of an existing record. This seems sensible to me

ok thanks @duncanjbrown that makes sense given the way this is going to be used, will get that change made now.

**Pull request merged**